docs/previous-versions/zustand-v3-create-context.md

```diff
@@ -6,7 +6,7 @@ nav: 18
 A special `createContext` is provided since v3.5,
 which avoids misusing the store hook.
 
-> **Note**: This function will likely be deprecated in v4 and removed in v5.
+> **Note**: This function will be deprecated in v4 and removed in v5.
```

Comment:
Does this `createContext` function miss-use React APIs? Or is it a performance aspect? Or do you simply not want to manage this code anymore?

Thank you.

Reply:
In v3, we only have one `create` function that creates a React hook.
Passing it via context (or props) can violate the rules of hooks. To avoid misusing it, `createContext` in v3.5 is carefully implemented so that it can't break the rules of hooks.

In v4, we provide a new general hook `useStore` along with `createStore` to create a vanilla store without React hooks.
These APIs are more React-ish, and recommended to use over `zustand/context` workaround. There's no risks to violate the rules of hooks. As we don't want to teach two usages, `zustand/context` is being deprecated and will be removed in v5.

So, the short answer is it's no longer necessary.

You are absolutely fine to keep it as a third-party package. The helper function could be useful for a pattern.

Reply:
so are you saying the migration path would be to call `createStore` and put the result into your own context? Can you maybe show how a replacement of zustand/context would look like?

Reply:
Good point. I answered it one of the discussions, but that should be in docs too.

Reply:
#1524